libraries/disp/viewers/helpers/rtfiffrawviewdelegate.cpp

```diff
@@ -550,53 +535,61 @@ void RtFiffRawViewDelegate::createPlotPath(const QModelIndex &index,
         }
     }
 
-    double dValue;
-    double dScaleY = option.rect.height()/(2*dMaxValue);
+    QPointF qSamplePosition;
 
+    double dValueScaled, dValue;
+    double dScaleY = option.rect.height()/(2*dMaxValue);
     double y_base = path.currentPosition().y();
-    QPointF qSamplePosition;
 
-    double dDx = ((float)option.rect.width()) / t_pModel->getMaxSamples();
+    // Calculate the smallest possible width for one sample data point
+    int iSkip = t_pModel->getMaxSamples() / option.rect.width();
```

Comment:
@gabrielbmotta Maybe we can do something similar in MNE Analyze?

Reply:
Yes, I tried doing something similar in mneanalyze,

```
    int iPaintStep = (int)(1.0/dDx) - 1;
    if (iPaintStep < 2){
        iPaintStep = 1;
    }

    for(int j = 0; j < data.size(); j = j + iPaintStep) {
        dValue = data[j] * dScaleY;

        //Reverse direction -> plot the right way
        newY = y_base - dValue;

        qSamplePosition.setY(newY);
        qSamplePosition.setX(path.currentPosition().x() + (dDx * (float)iPaintStep));
        path.lineTo(qSamplePosition);
    }
```

Reply:
any problems with the scroll bar, spacers, etc?

Reply:
none yet. I also changed how the spacers were being drawn, before we were wasting time looping through spacers that weren't getting displayed. Now we only draw enough to fill the screen. It seems to be a bit smoother than before with these two changes, I'm still looking for 
ways to streamline it more.